examples/vscode.js

```diff
@@ -0,0 +1,122 @@
+const { example } = require("../src/helpers");
+
+const { selectors: monacoSelectors } = require("./shared/monaco");
+
+const selectors = {
+  activityBar: {
+    root: ".activitybar",
+    menu: {
+      get root() {
+        return `${selectors.activityBar.root} [role="menubar"]`;
+      },
+      get button() {
+        return `${this.root} .menubar-menu-button[aria-label = "Application Menu"]`;
+      },
+      itemByLabel(label) {
+        return `.monaco-menu li.action-item [aria-label = "${label}"]`;
+      },
+    },
+    get items() {
+      return `${this.root} li.action-item`;
+    },
+    itemByTitle(title) {
+      return `${this.items}[title *= "${title}"]`;
+    },
+  },
+  sideBar: {
+    root: ".sidebar",
+    get explorer() {
+      return `*css=${this.root} >> h2:text("Explorer")`;
+    },
+    get listItems() {
+      return `${this.root} [role="treeitem"]`;
+    },
+    get selectedListItems() {
+      return `${this.listItems}.selected`;
+    },
+    listItemByTitle(title) {
+      return `${this.listItems} [title *= "${title}"]`;
+    },
+  },
+  editor: {
+    root: ".editor",
+    get tabs() {
+      return `${this.root} [role="tablist"] [role="tab"]`;
+    },
+    tabByTitle(title) {
+      return `${this.tabs}[title *= "${title}"]`;
+    },
+  },
+  search: {
+    root: ".search-view",
+    get input() {
+      return `${this.root} .search-container textarea`;
+    },
+    get replace() {
+      return `${this.root} .replace-container textarea`;
+    },
+  },
+  quickInput: {
+    root: ".quick-input-widget",
+    get save() {
+      return `${this.root} .quick-input-action >> text="OK"`;
+    },
+  },
+};
+
+const selectFile = async (path, page, doubleClick = false) => {
+  await page[doubleClick ? "dblclick" : "click"](
+    selectors.sideBar.listItemByTitle(path)
+  );
+  await page.waitForSelector(selectors.editor.tabByTitle(path));
+  // TODO: occasionallly, the wrong file is selected so we wait a bit to mitigate this
+  await page.waitForTimeout(250);
+};
+
+example("Visual Studio Code", async (page, { action }) => {
+  await action("Open Editor", async () => {
+    await page.goto("https://vscode-web-test-playground.azurewebsites.net/");
+
+    await page.click('text="Click to Continue"');
+
+    await Promise.all([
+      page.waitForSelector(".monaco-workbench"),
+      page.waitForSelector(selectors.sideBar.selectedListItems),
+      page.waitForSelector(selectors.editor.tabByTitle("large.ts")),
+    ]);
+  });
+
+  await action("Open files", async () => {
+    await selectFile("~/sample-folder/file.css", page);
+    await selectFile("~/sample-folder/file.html", page);
+    await selectFile("~/sample-folder/file.js", page, true);
+    await selectFile("~/sample-folder/file.md", page, true);
+  });
+
+  await action("Search files", async () => {
```

Comment:
I still worry about this really big examples, but i guess that's something we can change over time. The important thing is that this is really understandable

Reply:
I think we'll want to decompose these as they grow so that you can run all or part of them. Part of the challenge will be that some examples require some bootstrapping (like the "click to continue" block here) which will be useful to have consolidated. Thinking `beforeAll` sort of testing logic.